packages/listr2/src/listr.ts

```diff
@@ -132,12 +134,24 @@ export class Listr<
   }
 
   /**
-   * Add tasks to current task list.
+   * Add tasks to current task list. It will run the tasks if the listr instance is running.
    *
    * @see {@link https://listr2.kilic.dev/task/task.html}
    */
-  public add (tasks: ListrTask<Ctx, ListrGetRendererClassFromValue<Renderer>> | ListrTask<Ctx, ListrGetRendererClassFromValue<Renderer>>[]): void {
-    this.tasks.push(...this.generate(tasks))
+  public add (tasks: ListrTask<Ctx, ListrGetRendererClassFromValue<Renderer>> | ListrTask<Ctx, ListrGetRendererClassFromValue<Renderer>>[]): void | Promise<unknown> {
+    const newTasks = this.addTasks(tasks)
+
+    if (!this.running) {
+      return
+    }
+
+    if (this.isSubtask()) {
+      this.parentTask.emit(ListrTaskEventType.SUBTASK, newTasks)
+    }
+
+    return Promise.all(newTasks.map((task) => task.check(this.ctx))).then(() => {
+      return newTasks.map((task) => this.concurrency.add(() => this.runTask(task)))
+    })
```

Comment:
Is it not possible here to use the existing mechanism of `addTasks` or maybe consalidate these two things into somewhat same purpose.